src/config/field-overrides/relationshipType.js

```diff
@@ -117,28 +198,83 @@ class Constraint extends Component {
             },
         });
 
-        this.setState({
-            embeddedValue: relationshipConstraint,
-        });
+        this.setState(state => ({
+            selected: {
+                // Keep reference in state to check for selected programType etc
+                ...state.selected,
+                [modelType]: value,
+            },
+        }));
+    };
+
+    hasSelectedTrackerProgram = () =>
+        !!this.state.selected &&
+        !!this.state.selected.program &&
+        this.state.selected.program.programType === 'WITH_REGISTRATION';
```

Comment:
Why are you casting these to booleans? Existing object are also thruthy...

Let's hope JavaScript gets [optional chaining](https://github.com/tc39/proposal-optional-chaining) so we don't have to write conditions like this anymore :-(